x/evm/types/state_transition.go

```diff
@@ -106,25 +115,42 @@ func (st StateTransition) TransitionCSDB(ctx sdk.Context) (*big.Int, sdk.Result)
 	// Generate bloom filter to be saved in tx receipt data
 	bloomInt := big.NewInt(0)
 	var bloomFilter ethtypes.Bloom
+	var logs []*ethtypes.Log
 	if st.THash != nil && !st.Simulate {
-		logs := csdb.GetLogs(*st.THash)
+		logs, err = csdb.GetLogs(*st.THash)
+		if err != nil {
+			// TODO: handle error
+			logs = nil
+		}
 		bloomInt = ethtypes.LogsBloom(logs)
 		bloomFilter = ethtypes.BytesToBloom(bloomInt.Bytes())
 	}
 
 	// Encode all necessary data into slice of bytes to return in sdk result
-	returnData := EncodeReturnData(addr, bloomFilter, ret)
+	res := &ResultData{
+		Address: addr,
+		Bloom:   bloomFilter,
+		Logs:    logs,
+		Ret:     ret,
+	}
+	resultData, err := EncodeResultData(res)
+	if err != nil {
+		// TODO: where are the ethermint error types?
+		//returnData.Result = sdk.Err()
```

Comment:
if someone could point me to all the available cosmos sdk error types, that would be much appreciated